### ISSUE TITLE
[dotnet-sdk-7.0.100-rtm.22510.17] Microsoft.JSInterop.JSException: Converting circular structure to JSON when opening a new tab

### ISSUE THREAD

**Issue by Junjun-zhao** [dotnet/aspnetcore#44521]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

We are doing Migration test for our Blazor Server tool from .NET 6.0 to .NET 7.0. When trying to open a new tab in blazor server app, old code that works for 6.0, doesn't work in 7.0
```
await JSRuntime.InvokeAsync<object> ("open", $"https://www.bing.com", "_blank"); // does not work on 7.0, works in 6.0

await JSRuntime.InvokeVoidAsync("open", $"https://www.bing.com", "_blank"); // works
```

### Expected Behavior

It opens a new tab.

### Steps To Reproduce

**Minimal Repro steps**
 (Demo attached: [Demo.zip](https://github.com/dotnet/aspnetcore/files/9775367/Demo.zip)):

1.Create a blazor server app (.NET 7)
2.Go to any page, for example counter page, on top inject JSRuntime.
  ```
@inject IJSRuntime JSRuntime
```
3. Add a button
  ``` 
<button class="btn btn-primary" @onclick="OpenTab">Open bing on a new tab</button>
```
4. Add this code for OpenTab
```
 private async Task OpenTab()
  {
   // await JSRuntime.InvokeVoidAsync("open", $"https://www.bing.com", "_blank"); //this works
   await JSRuntime.InvokeAsync<object> ("open", $"https://www.bing.com", "_blank"); //this does not work
   }
```
**Actual Result:**
In the page, when clicking second time, it doesn't open the tab.

### Exceptions (if any)

**Exception Details**:
```
Microsoft.JSInterop.JSException: Converting circular structure to JSON
    --> starting at object with constructor 'Window'
    --- property 'window' closes the circle
TypeError: Converting circular structure to JSON
    --> starting at object with constructor 'Window'
    --- property 'window' closes the circle
    at JSON.stringify (<anonymous>)
    at D (https://localhost:44324/_framework/blazor.server.js:1:5345)
    at https://localhost:44324/_framework/blazor.server.js:1:3541
   at Microsoft.JSInterop.JSRuntime.InvokeAsync[TValue](Int64 targetInstanceId, String identifier, Object[] args)
   at BlazorApp1.Pages.Index.OpenTab() in C:\Users\v-zhiyang\Documents\Visual Studio 2022\BlazorApp1\BlazorApp1\Pages\Index.razor:line 12
   at Microsoft.AspNetCore.Components.ComponentBase.CallStateHasChangedOnAsyncCompletion(Task task)
   at Microsoft.AspNetCore.Components.RenderTree.Renderer.GetErrorHandledTask(Task taskToHandle, ComponentState owningComponentState)
```

### .NET Version

dotnet-sdk-7.0.100-rtm.22510.17

### Anything else?

**Affected App**: AppCompat Test Pass Management (internal tool)
**OS**: Windows 10 Enterprise 21H2
**CPU**: X64
**.NET Build Number**: 7.0.100-rc.2.22477.23

**Verify Scenarios:**
1). Windows 10 Enterprise 21H2 +.NET 7.0.100-rc.2.22477.23): Fail
2). Windows 10 Enterprise 21H2 +dotnet-sdk-7.0.100-rtm.22510.17. Fail
3). Windows 10 Enterprise 21H2 +dotnet-sdk-7.0.100-preview.2.22074.5 Fail
4). Windows 10 Enterprise 21H2 +dotnet-sdk-7.0.100-preview.7.22377.5: Fail
5). Windows 10 Enterprise 21H2 +dotnet-sdk-6.0.110-win-x64: Pass
6). Windows 10 Enterprise 21H2 +dotnet-sdk-5.0.502-win-x64: Pass
7). Windows 10 Enterprise 21H2 +dotnet-sdk-7.0.100-alpha.1.22071.3: Pass

**dotnet info:**
```
.NET SDK:
 Version:   7.0.100-rtm.22510.17
 Commit:    574e6f90fc

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19044
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\7.0.100-rtm.22510.17\

Host:
  Version:      7.0.0-rtm.22507.1
  Architecture: x64
  Commit:       b8a4363ecd

.NET SDKs installed:
  7.0.100-rtm.22510.17 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 7.0.0-rtm.22504.29 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 7.0.0-rtm.22507.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 7.0.0-rtm.22507.2 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
```

@dotnet-actwx-bot @dotnet/compat

**Comment by javiercn** [comment]
@Junjun-zhao thanks for filing the issue.

I would be surprised that the code you mentioned worked in 6.0. Very likely it did not work but the exception was not correctly propagated back to the server (which I think we fixed in 7.0).

`window.open` returns an object that is not json serializable and we have not changed anything in this area. The recommendation here is to switch to `await JSRuntime.InvokeVoidAsync("open", $"https://www.bing.com", "_blank");` as you already did.

**Comment by Junjun-zhao** [comment]
@javiercn Thanks for your information. I will close this issue and use the recommend method to update the code.
